src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterFactory.cs

```diff
@@ -32,7 +34,38 @@ public abstract class JsonConverterFactory : JsonConverter
         /// <param name="options">The <see cref="JsonSerializerOptions"/> being used.</param>
         /// <returns>
         /// An instance of a <see cref="JsonConverter{T}"/> where T is compatible with <paramref name="typeToConvert"/>.
+        /// If <see langword="null"/> is returned, a <see cref="NotSupportedException"/> will be thrown.
         /// </returns>
-        public abstract JsonConverter CreateConverter(Type typeToConvert, JsonSerializerOptions options);
+        public abstract JsonConverter? CreateConverter(Type typeToConvert, JsonSerializerOptions options);
+
+        internal override JsonPropertyInfo CreateJsonPropertyInfo()
+        {
+            throw new InvalidOperationException();
```

Comment:
It will never occur unless manually invoked by reflection etc. I'll add a Debug.Assert(false)

Reply:
nit: Can you use `Debug.Fail(message)` instead of `Debug.Assert(false)` so it self-documents when this could occur (exactly what you said in the PR comment can be the message).